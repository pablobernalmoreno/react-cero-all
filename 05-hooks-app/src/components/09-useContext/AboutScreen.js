import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser({})}>
        Clear
      </button>
    </div>
  );
};
