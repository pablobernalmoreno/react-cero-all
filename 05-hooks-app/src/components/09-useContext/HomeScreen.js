import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>HomeScreen </h1>
      <h2>{JSON.stringify(user)}</h2>
      <hr />
    </div>
  );
};
