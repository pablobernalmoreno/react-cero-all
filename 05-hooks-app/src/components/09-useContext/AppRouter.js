import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Route path="/" component={HomeScreen} />
          <Route component={HomeScreen} />
        </Switch>
      </div>
    </Router>
  );
};
