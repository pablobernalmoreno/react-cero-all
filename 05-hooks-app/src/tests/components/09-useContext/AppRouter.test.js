import { mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en AppRouter", () => {
  const user = {
    name: "Pablo",
    email: "pablioto@gmail.com",
  };

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser }}>
      <AppRouter />
    </UserContext.Provider>
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
