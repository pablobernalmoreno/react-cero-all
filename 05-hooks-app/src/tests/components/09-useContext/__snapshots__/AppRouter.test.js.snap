// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter debe de mostrarse correctamente 1`] = `
<AppRouter>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <NavBar>
          <nav>
            <ul>
              <li>
                <NavLink
                  to="./"
                >
                  <Link
                    aria-current="page"
                    className="active"
                    style={Object {}}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current="page"
                      className="active"
                      href="/"
                      navigate={[Function]}
                      style={Object {}}
                    >
                      <a
                        aria-current="page"
                        className="active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Home
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </li>
              <li>
                <NavLink
                  to="./about"
                >
                  <Link
                    aria-current={null}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/about",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      href="/about"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </li>
              <li>
                <NavLink
                  to="./login"
                >
                  <Link
                    aria-current={null}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/login",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      href="/login"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/login"
                        onClick={[Function]}
                      >
                        Login
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </li>
            </ul>
          </nav>
        </NavBar>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <HomeScreen
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div>
                <h1>
                  HomeScreen 
                </h1>
                <h2>
                  {"name":"Pablo","email":"pablioto@gmail.com"}
                </h2>
                <hr />
              </div>
            </HomeScreen>
          </Route>
        </Switch>
      </div>
    </Router>
  </BrowserRouter>
</AppRouter>
`;
